app/src/main/java/com/github/se/polyfit/ui/components/selector/DateSelector.kt

```diff
@@ -0,0 +1,110 @@
+package com.github.se.polyfit.ui.components.selector
+
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material3.DatePicker
+import androidx.compose.material3.DatePickerDialog
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.material3.TextButton
+import androidx.compose.material3.rememberDatePickerState
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.text.style.TextDecoration
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.github.se.polyfit.ui.theme.PrimaryPink
+import com.github.se.polyfit.ui.theme.PurpleGrey40
+import java.time.LocalDate
+import java.time.ZoneOffset
+import java.time.format.DateTimeFormatter
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun DateSelector(onConfirm: (LocalDate) -> Unit, modifier: Modifier = Modifier) {
+  var showDatePicker by remember { mutableStateOf(false) }
+
+  // I'm *pretty sure* this should select the right time and zone, but if test flakes this is why
+  val initialMillis = LocalDate.now().atStartOfDay().toInstant(ZoneOffset.UTC).toEpochMilli()
+  val datePickerState = rememberDatePickerState(initialSelectedDateMillis = initialMillis)
+
+  val selectedDate =
+      LocalDate.ofEpochDay(datePickerState.selectedDateMillis!! / (1000 * 60 * 60 * 24))
+  val dateString = selectedDate.format(DateTimeFormatter.ISO_LOCAL_DATE)
+
+  fun onConfirm() {
+    val date = LocalDate.parse(dateString, DateTimeFormatter.ISO_LOCAL_DATE)
+    onConfirm(date)
+    showDatePicker = false
+  }
+
+  Column(modifier = modifier.testTag("DateSelector")) {
+    Text(
+        text = "Add to Date",
+        style = MaterialTheme.typography.titleLarge,
+        color = PurpleGrey40,
+        modifier = Modifier.padding(16.dp, 0.dp).testTag("Title"))
+    Box(
+        contentAlignment = Alignment.Center,
+        modifier = Modifier.fillMaxWidth().testTag("DateBox")) {
+          TextButton(
+              onClick = { showDatePicker = true },
+              modifier = Modifier.testTag("ChangeDateButton")) {
+                FormatDate(dateString)
+              }
+
+          if (showDatePicker) {
+            DatePickerDialog(
+                onDismissRequest = { showDatePicker = false },
+                confirmButton = { TextButton(onClick = { onConfirm() }) { Text(text = "Done") } },
```

Comment:
It would be a good idea to disable the button until a validly formatted date is entered.

Reply:
i think im just gonna remove writing it by keyboard, i didn't intend that to be a feature and it causes more issues than benefit ;P